### DISCUSSION TITLE
self-hosting action_link doesn't include SUPABASE_URL when sending magic link.

### DISCUSSION THREAD

**Discussion by mikelyndon** [supabase/supabase#4381] [Questions]
I'm doing some local testing of the self-hosting option with the docker containers. I'm attempting to send a magiclink email.

**The kong gateway is on localhost:8005.**
If I test `http://localhost:8005/auth/v1/admin/generate_link` with the correct headers in postman, the action_link in the response is, `/auth/v1/verify?token=0DmSrnzMFeYtD3hjhPgNng&type=magiclink&redirect_to=http://localhost:3000`

And if I call the function `supabase.auth.signIn({ email })` from my nextjs app using inbucket for testing the email includes a similar link `http://localhost:9000/auth/v1/verify?redirect_to=http%3A%2F%2Flocalhost%2F&token=SzgK18MDwaDZZ9sUcVk1xg&type=magiclink`
I'm surprised that it's including the inbucket route - http://localhost:9000

So, how do I get the magiclink to include the kong gateway? Am I missing an environment variable somewhere?

**Comment by mikelyndon** [comment]
I think I've answered my own question but I'd love confirmation from someone who maintains the repo.

In the supabase/gotrue repo the example.env contains the env var API_EXTERNAL_URL which doesn't seem to be native to gotrue.

I've added this to the auth service of the docker-compose.yml 
`environment:`
      `GOTRUE_API_HOST: 0.0.0.0`
      `GOTRUE_API_PORT: 9999`
     ` API_EXTERNAL_URL: http://localhost:${KONG_HTTP_PORT}`

Setting this to my kong API Gateway (localhost:8005) and starting up the containers I'm getting the expected links in test emails and with generate_link.

**Reply by benjamin-kirkbride** [reply]
Thanks so much for this! Extremely frustrating that a lot of essential functionality is not configured nor documented..

**Reply by Paulkit** [reply]
So is it possible for self-hosting using deeplink? I have been trying to redirect to my flutter app via action_link but not working.

**Comment by amexcompliance** [comment]
I am having the same issue when supabase sends email verification. I modified env variables through Portainer and  The trick above did not work. However, what is puzzling is when I click on the verification link, although it is inaccessible the verification on the Supabase Users table happens anyway. Anyone from supabase team can comment on how to set up email verification on the local machine?
